libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginGoogleFragment.java

```diff
@@ -165,27 +181,38 @@ public void onSocialChanged(OnSocialChanged event) {
             switch (event.error.type) {
                 // WordPress account exists with input email address, but not connected.
                 case USER_EXISTS:
+                    AppLog.d(T.MAIN, "GOOGLE SIGNIN: onSocialChanged - wordpress acount exists but not connected");
                     mAnalyticsListener.trackSocialAccountsNeedConnecting();
                     mLoginListener.loginViaSocialAccount(mGoogleEmail, mIdToken, SERVICE_TYPE_GOOGLE, true);
                     break;
                 // WordPress account does not exist with input email address.
                 case UNKNOWN_USER:
+                    AppLog.d(T.MAIN, "GOOGLE SIGNIN: onSocialChanged - wordpress acount doesn't exist");
                     mAnalyticsListener.trackSocialErrorUnknownUser();
-                    showErrorDialog(getString(R.string.login_error_email_not_found, mGoogleEmail));
+                    showError(getString(R.string.login_error_email_not_found, mGoogleEmail));
+                    break;
+                // Too many attempts on sending SMS verification code. The user has to wait before they try again
+                case SMS_CODE_THROTTLED:
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - sms code throttled");
+                    showError(getString(R.string.login_error_sms_throttled));
                     break;
                 // Unknown error.
                 case GENERIC_ERROR:
-                // Do nothing for now (included to show all error types) and just fall through to 'default'
+                    // Do nothing for now (included to show all error types) and just fall through to 'default'
                 default:
-                    showErrorDialog(getString(R.string.login_error_generic));
+                    AppLog.d(T.MAIN, "GOOGLE SIGNIN: onSocialChanged - unknown error");
+                    showError(getString(R.string.login_error_generic));
                     break;
             }
-        // Response does not return error when two-factor authentication is required.
-        } else if (event.requiresTwoStepAuth) {
+            // Response does not return error when two-factor authentication is required.
+        } else if (event.requiresTwoStepAuth || "sms".equals(event.notificationSent)) {
```

Comment:
Let's replace the hard-coded `"sms"` with a constant like `Login2FaFragment.TWO_FACTOR_TYPE_SMS`.